packages/logs/src/domain/assembly.ts

```diff
@@ -63,10 +72,40 @@ export function startLogsAssembly(
 }
 
 interface Rum {
-  getInternalContext: (startTime?: RelativeTime) => Context
+  getInternalContext?: (startTime?: RelativeTime) => Context | undefined
 }
 
+interface BrowserWindow {
+  DD_RUM?: Rum
+  DD_RUM_SYNTHETICS?: Rum
+}
+
+let logsSentBeforeRumInjectionTelemetryAdded = false
+
 export function getRUMInternalContext(startTime?: RelativeTime): Context | undefined {
-  const rum = (window as any).DD_RUM as Rum
-  return rum && rum.getInternalContext ? rum.getInternalContext(startTime) : undefined
+  const browserWindow = window as BrowserWindow
+
+  if (willSyntheticsInjectRum()) {
+    const context = getFromGlobal(browserWindow.DD_RUM_SYNTHETICS)
+    if (!context && !logsSentBeforeRumInjectionTelemetryAdded) {
+      logsSentBeforeRumInjectionTelemetryAdded = true
+      addTelemetryDebug('Logs sent before RUM is injected by the synthetics worker', {
+        testId: getSyntheticsTestId(),
+        resultId: getSyntheticsResultId(),
+      })
+    }
+    return context
+  }
+
+  return getFromGlobal(browserWindow.DD_RUM)
+
+  function getFromGlobal(rum?: Rum): Context | undefined {
+    if (rum && rum.getInternalContext) {
+      return rum.getInternalContext(startTime)
+    }
+  }
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:**
Do we interchangeably use `Global` and `Context`? If not, maybe might be worth renaming the function name to explicitly mention `Context`?

Also we can use TS `?` to remove the need for `if` checks
```suggestion
  const getFromGlobal = (rum?: Rum): Context | undefined => rum?.getInternalContext(startTime)
```
(Note: if using `const` link in the example, we would need to lift where this function is defined)

Reply:
* I agree that I could have been more explicit. I pushed a commit to address that

* I tend to avoid the using `?.` operator because I observed that it produces more code than necessary when transpiled. If we compare the two approaches, transpiled + minimized:

```
// current implementation
function n(n){if(n&&n.getInternalContext)return n.getInternalContext()}
// your proposal
var t=function(n){var t;return null===(t=null==n?void 0:n.getInternalContext)||void 0===t?void 0:t.call(n)};
```

It's not a strong rule yet, but still something to keep in mind.